### PR TITLE
Add bug report and enhancement issue templates

### LINKED ISSUE #721: GitHub issue template(s) to include "definition of done"
As per our most recent call, it'd be great if we could get better at clearly defining the desired outcome for an issue. Ultimately, for each issue, we want to definitively know: *can this issue be closed right now?* And if it's difficult to tell, it means that the issue's not defined well enough.

### PR DESCRIPTION
**Pull request by thanethomson**
Closes #721.

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+**Steps to reproduce**
+
+<!--
+
+Please provide step-by-step instructions as to how to reproduce the problem.
+
+-->
+
+**When can we consider this to be "fixed"?**
```
Comment (greg-szabo) [inline .github/ISSUE_TEMPLATE/bug-report.md:21]: I'm trying to be somewhat pedantic with the English language here - for better or worse.

This question starting with "when" == "at one time" for me when I first read it. The description clarifies it well, but maybe we should be more to the point.

Maybe:
```suggestion
**Describe how a resolution looks like**
```

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+<!--
+
+Please describe clear and practical acceptance criteria for this issue (i.e. a
+"definition of done"). Anyone in the team should be able to look at this issue
+and determine whether or not the bug's been fixed.
```
Comment (greg-szabo) [inline .github/ISSUE_TEMPLATE/bug-report.md:27]:
```suggestion
and determine whether or not the bug has been fixed.
```

The grammar "soldier" in me...

File: .github/ISSUE_TEMPLATE/enhancement.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+labels: enhancement
+---
+<!--
+
+Please describe the enhancement you would like to see implemented in
+tendermint-rs. Give as much context as possible.
+
+-->
+
+**When can we consider this to be "done"?**
```
Comment (greg-szabo) [inline .github/ISSUE_TEMPLATE/enhancement.md:13]: Same comment as above: clarifying "when".

```suggestion
**Describe how the feature looks like and works in practice**
```

To be honest, this second item might be redundant here. What's the difference between "describing the definition of done" and "describing the enhancement you would like to see implemented". If it's only the level of detail, maybe we can remove this second question and just add "Please be as detailed as you can" to the first one.

**Review by greg-szabo — DISMISSED** [review]
Technically looks good. We can talk about wording all day. :) I'm approving this and letting Thane choose to take my minor improvement ideas into account or disregard them.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 2 above)
Comment (thanethomson) [reply .github/ISSUE_TEMPLATE/bug-report.md:26]: I usually purposefully put the contractions in nowadays to indicate less formality (being _Informal_ and all 😜).

Comment (greg-szabo) [reply .github/ISSUE_TEMPLATE/bug-report.md:26]: Fair point.

**Review by greg-szabo — DISMISSED** [review]
Re-approving after some changes. I'm still wary of the questions but it was already proven that I'm splitting hair in some cases.

File: .github/ISSUE_TEMPLATE/enhancement.md (hunk 3 above)
Comment (melekes) [reply .github/ISSUE_TEMPLATE/enhancement.md:13]: agree w/ @greg-szabo here

**Review by xla — APPROVED** [review]
Looking very good and concise. The main thing I find confusing is that feature requests and enhancements are interleaved. Enhancements usually result in smaller quality of life changes where feature requests can be entirely new sets of APIs, crates or functionality.

🌋 🎊 🍵 🐻

**Comment by thanethomson** [comment]
> The main thing I find confusing is that feature requests and enhancements are interleaved.

I think we could add that differentiation in future if incoming issues are too complex and need further sub-categorization. Right now, I see two broad "classes" of issues:

1. Something that should be working isn't working as expected.
2. We need to improve the system in some way.

File: .github/ISSUE_TEMPLATE/bug-report.md (hunk 1 above)
Comment (thanethomson) [reply .github/ISSUE_TEMPLATE/bug-report.md:20]: Is the comment below the heading not clear enough as to what the heading means?

File: .github/ISSUE_TEMPLATE/enhancement.md (hunk 3 above)
Comment (thanethomson) [reply .github/ISSUE_TEMPLATE/enhancement.md:13]: Same as the earlier comment - is the comment below the heading not clear enough here?

**Review by greg-szabo — APPROVED** [review]

**Pull request merged**